src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+- orange/blue - 0 bosons (|0,±>)
+- red/green - 1 boson (|1,±>)
+- brown/purple - 2 bosons (|2,±>)
+- grey/pink - 3 bosons (|3,±>)
+- blue/yellow - 4 bosons (|4,±>)
+
+Let's take the orange (|0,+>) and green (|1,->) lines. As we increase $\Delta E$, The energy of |0,+> goes up  and |1,-> goes down. Eventually, these levels end up with the same energy (they cross) despite having a different number of bosons - this is what allows the |0,+> to couple strongly to |1,-> i.e. it's what makes possible the spontaneous emission we saw in the last tutorial. 
+
+What's particularly interesting is the location of the crossings. For example, the crossing we just described happens when $\Delta E = \omega = 1$. This is physically significant because it is identical to the resonance condition we discovered in [Tutorial 2](https://nbviewer.jupyter.org/github/project-ida/two-state-quantum-systems/blob/master/02-perturbing-a-two-state-system.ipynb#Resonance) when the driving frequency of the classical field matched the transition frequency of the TSS. 
+
+In Tutorial 2, we found this resonance somewhat by accident. Now though, by looking at the other crossings in Fig 1 we can already start to predict the existence of other resonances when $\Delta E = 2\omega, 3\omega $ etc (we might call these "non-primary" resonances). We don't yet know what will happen at those resonances (maybe nothing at all 🤷) but it is not unreasonable to expect that when we switch the interaction on ($U\neq 0$) there will be some kind of interaction and we'll see the formation of anti-crossings similar to Tutorial 2.
+
+In essence, we are now going to try and think about these crossings as if they are independent two state systems whose states and couplings we have to determine.
+
+Let's see how we get on.
+
+
+
+### 4.2.2 - Crossings and anti-crossings
+
+We're now going to switch on the interaction term in the Hamiltonian. In order to more clearly visualise things, we will choose a stronger coupling than we did in the previous tutorial - specifically $U=0.2$.
+
+```python
+df = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[-0.5,5.5],legend=True, 
+        title=f"Stationary states for {H_latex}     ($\omega=1$, $U=0.2$)     (Fig 2)");
+plt.ylabel("Energy");
+```
+
+There are many things to say about what we see in Fig 2.
+
+Two main features are:
+1. the splitting of the levels increases with increasing level number i.e increasing number of bosons. 
+2. not all crossings have been split apart into anti-crossings as we expected - this means some levels don't interact with each other at all. 
```

Comment:
Can you maybe explain briefly which ones don't interact? Aren't the ones with no obvious splitting still avoided crossings that allow for interaction?

Reply:
I think this is possibly related to the question of the colour of lines, but I'm going to expand the paragraph below it a bit